FWCore/Framework/interface/WorkerInPath.h

```diff
@@ -109,7 +109,7 @@ namespace edm {
                                     typename T::TransitionInfoType const& info,
                                     ServiceToken const& token,
                                     StreamID streamID,
-                                    typename T::Context const* context) {
+                                    typename T::Context const* context) noexcept {
```

Comment:
Should we also mark `Worker::doWorkAsync` and `Worker::doWorkNoPrefetchingAsync` as `noexcept` as this routine calls them?

Reply:
As we discussed, I opened an issue about that (seemed too deep of a rabbit hole to be done as part of this PR).